mmcv/transforms/processing.py

```diff
@@ -498,6 +505,47 @@ def _crop_seg_map(self, results: dict, bboxes: np.ndarray) -> None:
             img = mmcv.imcrop(results['gt_semantic_seg'], bboxes=bboxes)
             results['gt_semantic_seg'] = img
 
+    def _crop_bboxes(self, results: dict, bboxes: np.ndarray) -> None:
+        """Update bounding boxes according to CenterCrop.
+
+        Args:
+            results (dict): Result dict contains the data to transform.
+            bboxes (np.ndarray): Shape (4, ), location of cropped bboxes.
+        """
+        if 'gt_bboxes' in results:
+            offset_w = bboxes[0]
+            offset_h = bboxes[1]
+            bbox_offset = np.array([offset_w, offset_h, offset_w, offset_h])
+            # gt_bboxes has shape (num_gts, 4) in (tl_x, tl_y, br_x, br_y)
+            # order.
+            gt_bboxes = results['gt_bboxes'] - bbox_offset
+            if self.clip_object_border:
+                gt_bboxes[:, 0::2] = np.clip(gt_bboxes[:, 0::2], 0,
+                                             results['img'].shape[1])
+                gt_bboxes[:, 1::2] = np.clip(gt_bboxes[:, 1::2], 0,
+                                             results['img'].shape[0])
+            results['gt_bboxes'] = gt_bboxes
+
+    def _crop_keypoints(self, results: dict, bboxes: np.ndarray) -> None:
+        """Update key points according to CenterCrop.
+
+        Args:
+            results (dict): Result dict contains the data to transform.
+            bboxes (np.ndarray): Shape (4, ), location of cropped bboxes.
+        """
+        if 'gt_keypoints' in results:
+            offset_w = bboxes[0]
+            offset_h = bboxes[1]
+            keypoints_offset = np.array([offset_w, offset_h, 0])
+            # gt_keypoints has shape (N, NK, 3) in (x, y, visibility) order,
+            # NK = number of points per object
+            gt_keypoints = results['gt_keypoints'] - keypoints_offset
+            gt_keypoints[:, :, 0] = np.clip(gt_keypoints[:, :, 0], 0,
```

Comment:
Should we decide whether to clip border based on self.clip_object_border:?

Reply:
I'm not sure. @ly015 may take a look.

Reply:
I suggest we remove the keypoint outside the border (mark it as invalid/invisible) instead of clipping the coordinate values.